src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -148,16 +151,28 @@ evalKEq true tools substitutionSimplifier _ (valid :< app) =
     Valid { patternSort } = valid
     Application { applicationChildren } = app
     evalEq t1 t2 = do
-        (result, _proof) <- makeEvaluate tools substitutionSimplifier ep1 ep2
-        case () of
-            _ | OrOfExpandedPattern.isTrue result ->
-                purePatternAxiomEvaluator (Bool.asInternal patternSort true)
-              | OrOfExpandedPattern.isFalse result ->
-                purePatternAxiomEvaluator (Bool.asInternal patternSort false)
-              | otherwise -> notApplicableAxiomEvaluator
-      where
-        ep1 = ExpandedPattern.fromPurePattern t1
-        ep2 = ExpandedPattern.fromPurePattern t2
+        let (expr, _proof) = Or.simplifyEvaluated
+                (MultiOr
+                    [ Predicated
+                        (Bool.asInternal patternSort true)
```

Comment:
If you use field names here it will be easier to follow what these three arguments mean.